deployment/data-streams/changeset/jd_revoke_jobs.go

```diff
@@ -0,0 +1,46 @@
+package changeset
+
+import (
+	"errors"
+	"fmt"
+
+	jobv1 "github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+var _ deployment.ChangeSetV2[CsRevokeJobSpecsConfig] = CsRevokeJobSpecs{}
+
+type CsRevokeJobSpecsConfig struct {
+	JobID string
+}
+
+type CsRevokeJobSpecs struct{}
+
+func (CsRevokeJobSpecs) Apply(e deployment.Environment, cfg CsRevokeJobSpecsConfig) (deployment.ChangesetOutput, error) {
+	resp, err := e.Offchain.RevokeJob(e.GetContext(), &jobv1.RevokeJobRequest{
+		IdOneof: &jobv1.RevokeJobRequest_Id{
+			Id: cfg.JobID,
+		},
+	})
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to revoke job: %w", err)
+	}
+
+	return deployment.ChangesetOutput{
+		Jobs: []deployment.ProposedJob{
```

Comment:
We might want to introduce new output to ChangesetOutput for RevokedJobs

Reply:
Would that new field/type provide any additional information about the revoked job? From my perspective the existing `Jobs` field is sufficient and the only confusing part is the name of its type - `ProposedJob` sounds like it only matches the result of `ProposeJob()`. But if we think about it as a job that was once proposed to the node, it makes a bit more sense.